src/components/fx/layout_attributes.js

```diff
@@ -18,6 +18,28 @@ fontAttrs.family.dflt = constants.HOVERFONT;
 fontAttrs.size.dflt = constants.HOVERFONTSIZE;
 
 module.exports = {
+    clickmode: {
+        valType: 'flaglist',
+        role: 'info',
+        flags: ['event', 'select'],
+        dflt: 'event',
+        editType: 'plot',
+        description: [
+            'Determines the mode of single click interactions.',
+            '*event* is the default value and emits the `plotly_click`',
+            'event. In addition this mode emits the `plotly_selected` event',
+            'in drag modes *lasso* and *select*, but with no event data attached',
+            '(kept for compatibility reasons).',
+            'The *select* flag enables selecting single',
+            'data points via click. This mode also supports persistent selections,',
+            'meaning that pressing Shift while clicking, adds to / subtracts from an',
+            'existing selection. *select* with `hovermode`: *x* can be confusing, consider',
+            'explicitly setting `hovermode`: *closest* when using this feature.',
+            'Selection events are sent accordingly as long as *event* flag is set as well.',
+            'When the *event* flag is missing, `plotly_click` and `plotly_selected`',
+            'events are not fired.'
+        ].join(' ')
+    },
```

Comment:
Probably want to add [`extras: ['none']`](https://github.com/plotly/plotly.js/blob/19f3800bac848e702948fcc32302ded56ea1cdb7/src/traces/scatter/attributes.js#L109) here as we [don't allow empty flaglists](https://github.com/plotly/plotly.js/blob/19f3800bac848e702948fcc32302ded56ea1cdb7/src/lib/coerce.js#L246)
Not sure why someone would want a click to do nothing, but as usual that's no reason to forbid it 😄

Reply:
> Does that mean, that someone can define `clickmode: 'none'`?

Exactly, then nothing at all will happen on a click. `extras` are single items you can use instead of the flag list - in some other places we have `'all'` for example, so you don't have to specify each individual piece. For the most part `'none'` should *just work* because it will fail all the `.indexOf(flag)` tests.

Reply:
Okay, so I'll just add `clickmode: 'none'`.